### PR TITLE
Ports TG Tackling and uses it to rework bloodsucker lunge

### PR DESCRIPTION
**Pull request by Arturlang**
## About The Pull Request
Ports the TG lunge PR which i cant credit because TG githubs down, plus uses the lunge component it adds to rework the bloodsucker lunge, to hopefully to be less bullshit.

## Why It's Good For The Game
The lunge component is very !!FUN!!, and should prove to be quite hilarious, and the bloodsucker lunge no longer being homing should make it less memes.

## Changelog
:cl: Ryll-Ryll
code: Adds the lunge component
add:Security vendors now have a premium item, the tackling gloves, which allow you to lunge at people, or walls, if you yknow, want to.
code: Ports the squish element from TG
add:All combat gloves have been replaced with guerrilla gloves, which allow tackling, but are otherwise the same functionally.
balance: Bloodsucker lunge now uses the tackling component
/:cl:

### CODE REVIEW

**Comment by necromanceranne** [comment]
Tackling was done by Shaps or Ryll over on tg.

File: code/datums/components/tackle.dm [hunk 1]
```diff
@@ -0,0 +1,482 @@
...
+	if(!(user.mobility_flags & MOBILITY_STAND))
+		to_chat(user, "<span class='warning'>You must be standing to tackle!</span>")
+		return
+
+	if(user.tackling)
+		to_chat(user, "<span class='warning'>You're not ready to tackle!</span>")
+		return
+
+	if(user.has_status_effect(STATUS_EFFECT_TASED)) // can't tackle if you just got shoved
+		to_chat(user, "<span class='warning'>You cant tackle while you are tased!</span>")
```
Comment (Ghommie) [inline code/datums/components/tackle.dm]:
```suggestion
		to_chat(user, "<span class='warning'>You can't tackle while tased!</span>")
```

File: code/modules/antagonists/bloodsucker/powers/fortitude.dm [hunk 2]
```diff
@@ -3,36 +3,36 @@
...
-	// Stop Running   (Taken from /datum/quirk/nyctophobia in negative.dm)
-	var/was_running = (user.m_intent == MOVE_INTENT_RUN)
+		fortitude_resist = max(0.3, 0.7 - level_current * 0.1)
+		H.physiology.brute_mod *= fortitude_resist
+		H.physiology.burn_mod *= fortitude_resist
+	was_running = (user.m_intent == MOVE_INTENT_RUN)
 	if(was_running)
 		user.toggle_move_intent()
-	while(bloodsuckerdatum && ContinueActive(user) || user.m_intent == MOVE_INTENT_RUN)
+		was_running = TRUE
```
Comment (Ghommie) [inline code/modules/antagonists/bloodsucker/powers/fortitude.dm]: `was_running` was already TRUE to begin with here.

File: code/datums/components/crafting/recipes.dm [hunk 3]
```diff
@@ -23,3 +23,12 @@
   */
 /datum/crafting_recipe/proc/check_requirements(mob/user, list/collected_requirements)
 	return TRUE
+
+/datum/crafting_recipe/gripperoffbrand
```
Comment (Ghommie) [inline code/datums/components/crafting/recipes.dm]: I don't think that's a good idea, we have no sticky tape and fingerless gloves are easy to get by.

File: code/modules/clothing/outfits/ert.dm [hunk 4]
```diff
@@ -69,7 +69,7 @@
 	id = /obj/item/card/id/ert/Security
 	suit = /obj/item/clothing/suit/space/hardsuit/ert/sec
 	glasses = /obj/item/clothing/glasses/hud/security/sunglasses
-	back = /obj/item/storage/backpack/security
+	gloves = /obj/item/clothing/gloves/tackler/combat/insulated
```
Comment (Ghommie) [inline code/modules/clothing/outfits/ert.dm]: Revert this line, all ERT officers already have guerilla gloves it seems.

File: code/modules/mob/living/carbon/carbon.dm [hunk 5]
```diff
@@ -98,6 +98,8 @@
 /mob/living/carbon/throw_impact(atom/hit_atom, datum/thrownthing/throwingdatum)
 	. = ..()
 	var/hurt = TRUE
+	if(src.GetComponent(/datum/component/tackler))
```
Comment (Ghommie) [inline code/modules/mob/living/carbon/carbon.dm]:
```suggestion
	if(GetComponent(/datum/component/tackler))
```

File: code/modules/research/designs/misc_designs.dm [hunk 6]
```diff
@@ -614,3 +614,27 @@
 	build_path = /obj/item/circuitboard/computer/sat_control
 	category = list("Computer Boards")
 	departmental_flags = DEPARTMENTAL_FLAG_ENGINEERING
+
+
+
+/////////////////////////////////////////
+////////////Tackle Gloves////////////////
```
Comment (Ghommie) [inline code/modules/research/designs/misc_designs.dm:621]: These designs are not included in any tech node, not that it matters that much.

**Review by Ghommie — CHANGES_REQUESTED** [review]

**Comment by Ghommie** [comment]
Also, reminder they restored normal combat gloves for nukies later on. https://github.com/optimumtact/-tg-station/commit/cd28642f35390d57d09a45c64485ae5a6eefb6b2

File: code/datums/components/crafting/recipes.dm (hunk 3 above)
Comment (Arturlang) [reply code/datums/components/crafting/recipes.dm]: Ill port the sticky tape i suppose

File: code/modules/research/designs/misc_designs.dm (hunk 6 above)
Comment (Arturlang) [reply code/modules/research/designs/misc_designs.dm:621]: Yeah, should i remove these from the code or just leave them in for whenever we get the BEPIS((hopefully someday))

File: code/datums/components/crafting/recipes.dm (hunk 3 above)
Comment (Arturlang) [reply code/datums/components/crafting/recipes.dm]: Just messed around with the recipy, see if its sufficent

**Review by Ghommie — APPROVED** [review]

**Pull request merged**